robotiq_joint_state_publisher/src/s_model_joint_states.cpp

```diff
@@ -0,0 +1,208 @@
+  mode = msg->gMOD;				// set mode
+
+  // if (mode == BASIC)        scissor = 137;
+  // else if (mode == PINCH)   scissor = 255;
+  // else if (mode == WIDE)    scissor = 0;
+  // else if (mode == SCISSOR) scissor = 0;
+
+  // Set all the joint values
+  joint_positions.at(0)  =  scissor_joint();
+  joint_positions.at(1)  =  finger_left.joint_1();
+  joint_positions.at(2)  =  finger_left.joint_2();
+  joint_positions.at(3)  =  finger_left.joint_3();
+  joint_positions.at(4)  = -joint_positions.at(0);
+  joint_positions.at(5)  =  finger_right.joint_1();
+  joint_positions.at(6)  =  finger_right.joint_2();
+  joint_positions.at(7)  =  finger_right.joint_3();
+  joint_positions.at(8)  =  finger_middle.joint_1();
+  joint_positions.at(9)  =  finger_middle.joint_2();
+  joint_positions.at(10) =  finger_middle.joint_3();
+}
+
+/**
+ * Main method.
+ */
+int main(int argc, char *argv[]) {
+
+  // Create Robotiq3
+  Robotiq3 robotiq;
+
+  // ROS init, nodehandle, and rate
+  ros::init(argc, argv, "s_model_joint_states");
+  ros::NodeHandle nh;
+  ros::Rate loop_rate(20);  // Hz
+
+  // joint state publisher
+  ros::Publisher joint_pub;
+  joint_pub = nh.advertise<sensor_msgs::JointState>("joint_states", 10);
+
+  // robotiq state message subscriber
+  ros::Subscriber joint_sub;
+  joint_sub = nh.subscribe("SModelRobotInput", 10, &Robotiq3::callback, &robotiq);
+
+  // wait until the URDF has been loaded to avoid publishing joints prematurely
+  while (!ros::isShuttingDown() && !nh.hasParam("/robot_description")) {
+    ROS_ERROR("[s_model_joint_states] Waiting for URDF");
+    ros::Duration(0.1).sleep();
+  }
+
+  // set pseudo-namespace
+  std::string gripper_prefix;
+  if (argc > 1) gripper_prefix = argv[1];
+  else gripper_prefix = "";
+
+  // joint names for sensor_msgs::JointState message
+  // order matters!
+  std::vector<std::string> joint_names(11, "");
+  joint_names.at(0).assign(gripper_prefix + "palm_finger_1_joint");
+  joint_names.at(1).assign(gripper_prefix + "finger_1_joint_1");
+  joint_names.at(2).assign(gripper_prefix + "finger_1_joint_2");
+  joint_names.at(3).assign(gripper_prefix + "finger_1_joint_3");
+  joint_names.at(4).assign(gripper_prefix + "palm_finger_2_joint");
+  joint_names.at(5).assign(gripper_prefix + "finger_2_joint_1");
+  joint_names.at(6).assign(gripper_prefix + "finger_2_joint_2");
+  joint_names.at(7).assign(gripper_prefix + "finger_2_joint_3");
+  joint_names.at(8).assign(gripper_prefix + "finger_middle_joint_1");
+  joint_names.at(9).assign(gripper_prefix + "finger_middle_joint_2");
+  joint_names.at(10).assign(gripper_prefix + "finger_middle_joint_3");
+
+  // std::cout << "GRIPPER JOINT NAMES: ";
+  // std::copy(joint_names.begin(), joint_names.end(), std::ostream_iterator<std::string>(std::cout, "\n"));
+
+  // Output JointState message
+  sensor_msgs::JointState joint_msg;
+  // Joint names to JointState message
+  joint_msg.name = joint_names;
+
+  while (!ros::isShuttingDown()) {
```

Comment:
I normally use `ros::ok()`, reads better.  Is there a reason to use `isShuttingDown` (just curious, don't have to fix).